src/Copy_JIRA_info_for_Git.user.js

```diff
@@ -1,241 +1,573 @@
+            type.selected = false;
+          });
+          selectedType.selected = true;
+          this.setConstantPart();
 
-    let container = document.createElement('div');
-    container.classList.add('btn-group', 'copyContainer');
+          if (getBranches && previousSelected !== selectedType) {
+            const selectedPlatform = this.getSelected(this.platforms);
+            const selectedIssueType = this.getSelected(this.issueTypes);
 
-    if ((localStorage.getItem('JiraInfoForGit') && localStorage.getItem('JiraInfoForGit') === 'closed')) {
-        container.classList.add('hide');
+            if (selectedPlatform && selectedIssueType) {
+              this.getBranches(selectedPlatform.key, selectedIssueType.filter);
+            }
+          }
+        },
+        setCommitMessage() {
+          const ticket = this.getTicketID();
+          const ticketName = this.getTicketName();
+          const ticketNameFormatted = ticketName.replace(/\s\s/ig, ' ').replace(/\[.*?\]/ig, '').trim();
+          this.commitMessage = `[${ticket}] ${ticketNameFormatted}`;
+        },
+        setConstantPart() {
+          const ticket = this.getTicketID();
+          const issueType = this.issueTypes.filter(issue => issue.selected)[0];
+          const brand = this.branches.filter(branch => branch.selected)[0];
+
+          if (!brand || !issueType) {
+            this.showBranchName = false;
+            return;
+          }
+
+          this.showBranchName = true;
+          this.constantPart = `${issueType.key}/${brand.label}/${ticket}-`;
+          this.checkBranchExists();
+        },
+        getTicketName() {
+          return document.querySelector('#summary-val').innerText;
+        },
+        getTicketID() {
+          return document.querySelector('#key-val').innerText;
+        },
+        getEditablePart() {
+          const ticketName = this.getTicketName();
+          return this.formatToValidBranchName(ticketName);
+        },
+        formatToValidBranchName(string, limit = 5) {
+          let formattedString = string.replace(/["'`]/g, '') /* remove quotes */
+                                      .replace(/\s-\s/g, '') /* remove dashes */
+                                      .replace(/\s\s/ig, ' ') /* Replace double spaces with single space */
+                                      .replace(/\[.*?]/ig, '') /* Strip tags in square brackets e.g.: [tag] */
+                                      .trim() /* After removing square brackets - there might be leading white space left, so need to trim */
+                                      .replace(/\s/ig, '-') /* Replace all spaces with dashes */
+                                      .replace(/^[./]|\.\.|@{|[\/.]$|^@$|[~^:\x00-\x20\x7F\s?*[\]\\]/ig, ''); /* Strip all forbidden chars */
+          return limit ? formattedString.split('-').slice(0, limit).join('-') : formattedString;
+        },
+        getBranches(type, search) {
+          const params = [
```

Comment:
@D4ST1N please move everything related to actual GitLab integration to separate GitLab-util class

Reply:
@TheBit done